File: OIB/lab_2/pserand/pserand.java
import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;

public class pserand {
    private static final int N = 128;
    private static final long SEED = 11342;
    private static final String OUTPUT_FILENAME = "seq_java.txt";

    public static void main(String[] args) {
        Random random = new Random(SEED);
        try {
            FileWriter writer = new FileWriter(OUTPUT_FILENAME);
            for (int i = 0; i < N; i++) {
                int bit = random.nextInt(2); 
                writer.write(String.valueOf(bit));
            }
            writer.close();
        } catch (IOException e) {
            System.err.println("Ошибка ввода/вывода при записи файла: " + e.getMessage());
        }
    }
}

File: OIB/lab_2/pserand/pserand.py
import random

def generate_and_write_sequence(n: int, seed: int, output_path: str) -> None:
    random.seed(seed)
    
    sequence_bits = []

    for _ in range(n):
        bit = random.getrandbits(1)
        sequence_bits.append(str(bit))

    try:
        with open(output_path, 'w') as f:
            f.write("".join(sequence_bits))
    except IOError as e:
        print(f"Ошибка при записи файла: {e}")

if __name__ == "__main__":
    N = 128
    SEED = 11342 
    OUTPUT_FILE = "seq_py.txt"

    generate_and_write_sequence(N, SEED, OUTPUT_FILE)